tests/integration/conftest.py

```diff
@@ -180,14 +181,178 @@ async def deploy_model(
 @pytest_asyncio.fixture(scope="module")
 async def kubernetes_cluster(request: pytest.FixtureRequest, ops_test: OpsTest):
     """Deploy local kubernetes charms."""
-    model = "main"
+    cluster_model = "main"
     charm_path = ("worker/k8s", "worker")
     charms = [Charm(ops_test, Path("charms") / p) for p in charm_path]
     charm_files = await asyncio.gather(
         *[charm.resolve(request.config.option.charm_files) for charm in charms]
     )
+
     bundle = Bundle(ops_test, Path(__file__).parent / "test-bundle.yaml")
     for path, charm in zip(charm_files, charms):
         bundle.switch(charm.app_name, path)
-    async with deploy_model(request, ops_test, model, bundle) as the_model:
+
+    async with deploy_model(request, ops_test, cluster_model, bundle) as the_model:
```

Comment:
I think SOOOOO much of this needs to move to the `pytest-operator`.  Gonna evaluate and look at this tomorow

Reply:
Alright, sounds good!